backend/prompt_studio/prompt_studio/helper.py

```diff
@@ -6,27 +6,61 @@
 
 
 class PromptStudioHelper:
+    @staticmethod
+    def get_prompt_model(is_sps: bool):
+        """Dynamically import and return the appropriate prompt model.
+
+        Args:
+            is_sps (bool): Flag indicating whether to use the SPSPrompt model.
+
+        Returns:
+            model (class): The appropriate prompt model class
+            (ToolStudioPrompt or SPSPrompt).
+
+        Raises:
+            ImportError: If the SPSPrompt model could not be imported.
+        """
+        if is_sps:
+            try:
+                from pluggable_apps.simple_prompt_studio.sps_prompt.models import (
+                    SPSPrompt,
+                )
+
+                return SPSPrompt
+            except ImportError as e:
+                logger.error("SPSPrompt model could not be imported.")
+                raise ImportError("SPSPrompt model could not be imported.") from e
+        return ToolStudioPrompt
+
     @staticmethod
     def reorder_prompts_helper(
-        prompt_id: str, start_sequence_number: int, end_sequence_number: int
+        prompt_id: str,
+        start_sequence_number: int,
+        end_sequence_number: int,
+        is_sps: bool = False,
     ) -> list[dict[str, int]]:
         """Helper method to reorder prompts based on sequence numbers.
 
         Args:
             prompt_id (str): The ID of the prompt to be reordered.
             start_sequence_number (int): The initial sequence number of the prompt.
             end_sequence_number (int): The new sequence number of the prompt.
+            is_sps (bool): Flag to determine the prompt model to use.
 
         Returns:
             list[dict[str, int]]: A list of updated prompt data with their IDs
             and new sequence numbers.
         """
-        prompt_instance: ToolStudioPrompt = ToolStudioPrompt.objects.get(pk=prompt_id)
-        filtered_prompts_data = []
+
+        prompt_model = PromptStudioHelper.get_prompt_model(is_sps)
```

Comment:
If sps is a cloud only feature, there should be no mention of this in OSS code ideally - do you think we can properly make this class a plugin instead? How / when will `is_sps` be `True`?

Reply:
@chandrasekharan-zipstack We're using the same logic for reordering the prompts in both the main and simple prompt studio. The `is_sps` argument will be `True` if the request is being made from SPS. If you look at the `get_prompt_model` method in the same [file](https://github.com/Zipstack/unstract/blob/bc554ec5c3748f25057f05b86d1d3f7c1208ae43/backend/prompt_studio/prompt_studio/helper.py#L10), you'll see that we use the `is_sps` value to decide which model to use.

Reply:
@chandrasekharan-zipstack Thank you for the suggestion to eliminate the use of is_sps. I have made the necessary changes. Please review that part once more and address the comment.